spring-boot-autoconfigure/src/main/java/org/springframework/boot/autoconfigure/http/codec/CodecsAutoConfiguration.java

```diff
@@ -0,0 +1,39 @@
+/*
+ * Copyright 2012-2017 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.springframework.boot.autoconfigure.http.codec;
+
+import org.springframework.boot.autoconfigure.AutoConfigureAfter;
+import org.springframework.boot.autoconfigure.EnableAutoConfiguration;
+import org.springframework.boot.autoconfigure.condition.ConditionalOnClass;
+import org.springframework.boot.autoconfigure.jackson.JacksonAutoConfiguration;
+import org.springframework.context.annotation.Configuration;
+import org.springframework.context.annotation.Import;
+import org.springframework.http.codec.CodecConfigurer;
+
+/**
+ * {@link EnableAutoConfiguration Auto-configuration}
+ * for {@link org.springframework.core.codec.Encoder}s and {@link org.springframework.core.codec.Decoder}s.
+ * @author Brian Clozel
+ */
+@Configuration
+@ConditionalOnClass(CodecConfigurer.class)
+@AutoConfigureAfter(JacksonAutoConfiguration.class)
+@Import(JacksonCodecConfiguration.class)
```

Comment:
`JacksonCodecConfiguration` is not a lot of code. Perhaps we should move it as an inner class of this one?